src/main.js

```diff
@@ -1,8 +1,38 @@
 import { getPokemonList } from './api/index.js';
 import { createDropdownTypes, createPokemonList } from './dom/index.js';
 
-const addListeners = () => {
-	// Code goes here
+const filterCards = () => {
+	// Define variable to select the Dropdown from the DOM.
+	const dropdownType = document.getElementById("dropdown-types"); 
+
+	// Add the event listener to the Dropdown to catch when the selection changes and filter the list based on the selection
+	dropdownType.addEventListener('change', (event) => {
+		// Constant with the Card list item DOM element
+		const cardsList = document.querySelectorAll(".card-list__item");
+		// Variable to store the value selected in the dropdown
+		let currentSelection = dropdownType.value;
+		// Variable to store the type of the pokemon that is being evaluated
+		let currentPokemonType = "";
+		
+		// Logic to determine wheter the element should have the hide class or not
+		// If the selection is "all", the hide class should be removed from all the elements
+		// If the selection is a type, the "hide"class should be added to the pokemons that do not have that type
+		// and should be removed from the pokemons that have that type.
+		if (currentSelection == "all") {
+			cardsList.forEach(element => {
+					element.classList.remove("hide");
+			});
+		} else {
+			cardsList.forEach(element => {
+				currentPokemonType = element.attributes[0].value;
+				if (currentPokemonType.includes(currentSelection)) {
+						element.classList.remove("hide");
```

Comment:
Check the second attribute of the function `toggle`. It's a boolean that forces the addition or removal of the class based on its value.
![image](https://user-images.githubusercontent.com/11170707/124197377-03480a80-da94-11eb-8607-21bb76dd5290.png)